src/design_notes/variadic_generics_design.md

```diff
@@ -0,0 +1,133 @@
+
+# Variadic Generics
+
+A draft for Variadic Generics has existed since ~2013, and there have been multiple postponed RFCs surrounding
+the topic. Despite this, the difficulty of designing such a system as well as lack of singular best choice
+has led to no resolution. This document has been written as an attempt to 'sum up' proposals and discussion in
+this space.
+
+# Proposals so far
+
+Following will be an overview of each of the proposals so far, followed by their individual pros and cons
+
+## EddyB's Draft
+
+This is a simple proposal, and the oldest one. It proposes the basic idea that all variadic types can
+be seen as just a tuple with N elements. The syntax is thus built around allowing expansions of tuple
+types into their components, and taking multiple types that will be condensed into a tuple.
+
+For using these types, it is proposed that one destructure the tuple, generally into a single head, and the remaining
+tail. This matches the recursive style used in C++.
+
+### Pros
+
+- Simple. Doesn't add much syntax while allowing the driving use case, the `Fn` traits and similar designs.
+
+- Allows intuitive function call ergonomics
+
+### Cons
+
+- Mentioned by eddyb, a subtuple type could have different padding than its parent.
+  EG: `&(A, B, C, D) != &A, &(B, C, D)` 
+
+- The `..` syntax is already used in ranges, so some other syntax would be needed
```

Comment:
We have `...T` syntax (three dots) available now, since inclusive ranges now use `..=`.